File: lab2-functionalprimer/python3/func_as_arg.py
def listFunc():
    a = input("enter a start number")
    b = input("enter an end number")
    # return [i for i in range(a, b)] #Create list of ints from 1 to 5, Haskell equivalent [1..5]
    return list(range(a, b+1))

def applicatorFunc(inpFunc, s):
    if s=='s':
        return sum(inpFunc())
    else:
        return sum(inpFunc())/5

print(applicatorFunc(listFunc, 's'))

File: lab2-functionalprimer/javascript/func_as_arg.js
//Create list of ints from 1 to 5, Haskell equivalent [1..5]
function arrFunc(a,b){
    let arr = [];
    // let a = parseInt(prompt("Enter a start number"));
    // let b = parseInt(prompt("Enter an ending number"));
    for (let i = a; i<=b; i++) {
        arr.push(i);
    }
    return arr;    
}

function applicatorFunc(inpFunc, s,a,b){
    if(s=='s'){
        const arr = inpFunc(a,b);        
        let sum = arr.reduce((accumulator, currentValue) => accumulator + currentValue, 0);
        return sum;
    }
    else{        
        const arr = inpFunc(a,b);
        let sum = arr.reduce((accumulator, currentValue) => accumulator + currentValue, 0);
        return sum/(b-a+1);
    }
}

let x = applicatorFunc(arrFunc, 'a', 1, 10);
console.log(x);
